internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,604 @@
+			newSecretEnvVar("OCM_SERVICE_CLIENT_SECRET", "ocm-service.clientSecret", ocmSSOSecretName),
+			newSecretEnvVar("AWS_SECRET_ACCESS_KEY", "aws_secret_access_key", s3SecretName),
+			newSecretEnvVar("AWS_ACCESS_KEY_ID", "aws_access_key_id", s3SecretName),
+			newSecretEnvVar("S3_REGION", "aws_region", s3SecretName),
+			newSecretEnvVar("S3_BUCKET", "bucket", s3SecretName),
+			newSecretEnvVar("S3_ENDPOINT_URL", "endpoint", s3SecretName),
+			newSecretEnvVar("AWS_SECRET_ACCESS_KEY_PUBLIC", "aws_secret_access_key", publicS3SecretName),
+			newSecretEnvVar("AWS_ACCESS_KEY_ID_PUBLIC", "aws_access_key_id", publicS3SecretName),
+			newSecretEnvVar("S3_REGION_PUBLIC", "aws_region", publicS3SecretName),
+			newSecretEnvVar("S3_BUCKET_PUBLIC", "bucket", publicS3SecretName),
+			newSecretEnvVar("S3_ENDPOINT_URL_PUBLIC", "endpoint", publicS3SecretName),
+
+			// TODO: Should these be in a ConfigMap? the assisted-service-config CM?
+			newEnvVar("ISO_IMAGE_TYPE", "minimal-iso"),
+			newEnvVar("S3_USE_SSL", "false"),
+			newEnvVar("LOG_LEVEL", "info"),
+			newEnvVar("LOG_FORMAT", "text"),
+			newEnvVar("INSTALL_RH_CA", "false"),
+			newEnvVar("REGISTRY_CREDS", ""),
+			newEnvVar("AWS_SHARED_CREDENTIALS_FILE", "/etc/.aws/credentials"),
+			newEnvVar("DEPLOY_TARGET", "ocp"),
+			newEnvVar("STORAGE", "filesystem"),
+			newEnvVar("ISO_WORKSPACE_BASE_DIR", "/data"),
+			newEnvVar("ISO_CACHE_DIR", "/data/cache"),
+
+			// database
+			newSecretEnvVar("DB_HOST", "db.host", databaseSecretName),
+			newSecretEnvVar("DB_NAME", "db.name", databaseSecretName),
+			newSecretEnvVar("DB_PASS", "db.password", databaseSecretName),
+			newSecretEnvVar("DB_PORT", "db.port", databaseSecretName),
+			newSecretEnvVar("DB_USER", "db.user", databaseSecretName),
+
+			// image overrides
+			newEnvVar("AGENT_DOCKER_IMAGE", AgentImage()),
+			newEnvVar("CONTROLLER_IMAGE", ControllerImage()),
+			newEnvVar("INSTALLER_IMAGE", InstallerImage()),
+			newEnvVar("SELF_VERSION", ServiceImage()),
+
+			{
+				Name: "NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+		},
+		EnvFrom: []corev1.EnvFromSource{
+			{
+				ConfigMapRef: &corev1.ConfigMapEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: assistedServiceConfigMapName,
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "bucket-filesystem",
+				MountPath: "/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(2000*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
```

Comment:
@romfreiman does it end up with the test that you made for 1K clusters? (with agents and BMH it will probably double)